aten/src/ATen/native/cuda/Shape.cu

```diff
@@ -240,6 +270,27 @@ Tensor cat_cuda(TensorList inputs, int64_t dimension) {
   return out;
 }
 
+inline c10::MemoryFormat compute_output_memory_format(const TensorList &inputs) {
```

Comment:
can you do it with std::all_of?

Reply:
I don't think this is possible. `std::all_of` is for boolean values, but memory format has `ChannelsLast` and `ChannelsLast3d`, we need to preserve both case.

Reply:
Cool, I forgot about `ChannelsLast3d`.